### PR TITLE
Remove excessive logs in LogUtils

### PR DESCRIPTION
**Pull request by SadiJr**
### Description

ACS has one method `(LogUtils#logGsonWithoutException)` which tries to transform one object to JSON, to write this object in the logs. If this tranformation fails, the message `error to decode` is written in DEBUG, which in some scenarios can cause a lot of useless messages. This method was refactored, and this message was changed to write only in TRACE.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
It was tested in a local lab:

1. I performed some basic operations in VMs, like creating new VMs, destroying these VMs, migrating volumes, and so on;
2. Before, there were a lot of messages `error to decode` like, which didn't provide enough information to the operators;
3. This message is no longer displayed in the DEBUG level.

### CODE REVIEW

File: utils/src/main/java/com/cloud/utils/LogUtils.java [hunk 1]
```diff
@@ -78,21 +79,29 @@ public static Set<String> getLogFileNames() {
...
         List<String> gsons = new ArrayList<>();
         for (Object object : objects) {
             try {
                 gsons.add(GSON.toJson(object));
             } catch (Exception e) {
-                LOGGER.debug(String.format("Failed to log object [%s] using GSON.", object != null ? object.getClass().getSimpleName() : "null"));
-                gsons.add("error to decode");
+                Object errObj = ObjectUtils.defaultIfNull(object, "null");
+                LOGGER.trace(String.format("Failed to log object [%s] using GSON.", errObj.getClass().getSimpleName()));
+                gsons.add(errObj.toString());
```
Comment (DaanHoogland) [inline utils/src/main/java/com/cloud/utils/LogUtils.java:97]:
```suggestion
                gsons.add("error decoding " + errObj.toString());
```

File: utils/src/main/java/com/cloud/utils/LogUtils.java [hunk 2]
```diff
@@ -78,21 +79,29 @@ public static Set<String> getLogFileNames() {
...
             }
         }
         try {
             return String.format(formatMessage, gsons.toArray());
         } catch (Exception e) {
             String errorMsg = String.format("Failed to log objects using GSON due to: [%s].", e.getMessage());
-            LOGGER.error(errorMsg, e);
+            LOGGER.trace(errorMsg, e);
```
Comment (DaanHoogland) [inline utils/src/main/java/com/cloud/utils/LogUtils.java:104]: I think this is a "WARN" at least.

Comment (SadiJr) [reply utils/src/main/java/com/cloud/utils/LogUtils.java:104]: @DaanHoogland As this message occurs when an object is not being converted correctly to JSON (happens quite often with VMWare SDK objects) and does not mean an exception or error/warning in the environment, I think putting the log as warning might unnecessarily aware the operators.

Comment (DaanHoogland) [reply utils/src/main/java/com/cloud/utils/LogUtils.java:104]: fair enough

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: utils/src/main/java/com/cloud/utils/LogUtils.java [hunk 3]
```diff
@@ -78,21 +79,29 @@ public static Set<String> getLogFileNames() {
         return fileNames;
     }
 
+    /**
+     * Tries to convert message parameters to JSON format and use them in the message.
+     * @param formatMessage message to format.
+     * @param objects objects to convert to JSON. An null object will be defaulted to the String "null";
```
Comment (stephankruggg) [inline utils/src/main/java/com/cloud/utils/LogUtils.java:85]:
```suggestion
     * @param objects objects to convert to JSON. A null object will be defaulted to the String "null";
```

**Review by ezntt — APPROVED** [review]

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: utils/src/main/java/com/cloud/utils/LogUtils.java (hunk 3 above)
Comment (SadiJr) [reply utils/src/main/java/com/cloud/utils/LogUtils.java]: Done, thanks.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]

**Review by RodrigoDLopez — APPROVED** [review]
lgtm.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**